plugins/batterystat.koplugin/main.lua

```diff
@@ -64,59 +63,64 @@ function Usage:percentagePerHour()
 end
 
 function Usage:remainingHours()
+    if self:percentagePerHour() == 0 then return "n/a" end
     local curr = State:new()
     return curr.percentage / self:percentagePerHour()
 end
 
 function Usage:chargingHours()
+    if self:percentagePerHour() == 0 then return "n/a" end
     local curr = State:new()
-    return (curr.percentage - 100) / self:percentagePerHour()
+    return math.abs(curr.percentage - 100) / self:percentagePerHour()
 end
 
 local function shorten(number)
+    if number == "n/a" then return _("n/a") end
     return string.format("%.2f", number);
 end
 
 function Usage:dump(kv_pairs)
-    table.insert(kv_pairs, {_("    Consumed %"), shorten(self.percentage)})
-    table.insert(kv_pairs, {_("    Total minutes"), shorten(self:minutes())})
-    table.insert(kv_pairs, {_("    % per hour"), shorten(self:percentagePerHour())})
+    table.insert(kv_pairs, {string.format("%s%s",INDENTATION, _("Consumed %")), shorten(self.percentage)})
+    table.insert(kv_pairs, {string.format("%s%s",INDENTATION, _("Total time")), util.secondsToHClock(self.time, true, true)})
+    table.insert(kv_pairs, {string.format("%s%s",INDENTATION, _("% per hour")), shorten(self:percentagePerHour())})
 end
 
 function Usage:dumpRemaining(kv_pairs)
-    table.insert(kv_pairs, {_("    Estimated remaining hours"), shorten(self:remainingHours())})
+    table.insert(kv_pairs, {string.format("%s%s",INDENTATION, _("Estimated remaining hours")), shorten(self:remainingHours())})
 end
 
 function Usage:dumpCharging(kv_pairs)
-    table.insert(kv_pairs, {_("    Estimated hours for charging"), shorten(self:chargingHours())})
+    table.insert(kv_pairs, {string.format("%s%s",INDENTATION, _("Estimated hours for charging")), shorten(self:chargingHours())})
```

Comment:
@poire-z Does this work for RTL?

Reply:
Possibly not - but go ahead with that as it is, it is done the same way in the System Statistics plugin:
<kbd>![sysstats_rtl](https://user-images.githubusercontent.com/24273478/69426105-ac614780-0d2c-11ea-8117-9727b58651b4.png)</kbd>

There are much more stuff than I expected that will need to be adapted for proper and code-clean RTL mirroring, and I have yet no idea how we'll go with these indented keys in Keyvaluepage...
We'll fix all these, either manually in each of them, or globally if we find a trick to have that done in KeyValuePage or even upper.

As an example, our lazy aligment tricks to use a different padding_left vs padding_right in FrameContainer - would need to be replaced by a HorizontalSpan and the FrameContainer in a HorizontalGroup, so HorizontalGroup is the only one that'd need to deal with mirroring.
My wish is that we should continue to just think LTR when developping, with some simple rules to follow for special cases (like chevron, or string bits to be BD.wrap() before concatenating them - and all should be done automagically by low level widgets when switching to some RTL language.

Reply:
What I mean is that lazy `   some spaces for alignment` can be translated however it needs to be in RTL if they're part of the string, as they are now. If this code would require extra code for inverting then this is a bad change.

Reply:
Note that a string like `الذاكرة الإجمالية (MB)` overlooked and left out the spaces, unless it was a conscious decision. This is a Transifex flaw.

Reply:
> left out the spaces.

Oh, right. If I add the leading spaces back in ar_AA/koreader.po, then it looks ok! So, no issue with RTL at having spaces at start for indentation (I thought the bidi algo might push all spaces at end, but it doesn't). So good.

Anyway, the way this PR does it (indentation outside of translated string) is fine. (The way System statistics plugin does it is not.)

Reply:
Right, but does "    ".."RTL string" actually put them at the start or at the end?

Reply:
At the start of the logical order of bytes. So it will be on the right (so at the RTL start) when text is rendered RTL (the bidi algo consider the spaces neutral, so all RTL when next real char is RTL).

There are many things the bidi algo (and/or xtexp.cpp :) do just right - we can still keep thinking LTR, and assume it will work ok with RTL.

Reply:
Alright, in that case this bit of programmatic cleverness should okay. But in matters of internationalization anything clever is suspect by default. :-)